packages/Sandblocks-Core/SBBlock.class.st

```diff
@@ -421,16 +422,24 @@ SBBlock >> completeSubtree [
 { #category : #suggestions }
 SBBlock >> completionSuggestionsFor: aString [
 
-	^ aString size > 2
+	^ (self suggestAlways or: [aString size > 2])
 		ifTrue: [ | baseClass |
 			baseClass := self guessClassExpensive.
 			baseClass isNil
-				ifFalse: [
-					Array streamContents: [:stream |
+				ifFalse: [ | allMatches fuzzyMatches exactMatches |
+					exactMatches := Array streamContents: [:stream |
+						baseClass withAllSuperclassesDo: [:class |
+							class selectors sorted do: [:sel |
+								(sel asString sandblockBeginsWith: aString) ifTrue: [stream nextPut: sel asString]]]].
+					fuzzyMatches := Array streamContents: [:stream |
+						baseClass withAllSuperclassesDo: [:class |
+							class selectors sorted do: [:sel |
+								(sel asString sandblockMatch: aString) ifTrue: [stream nextPut: sel asString]]]].
+					allMatches := Array streamContents: [:stream |
 						baseClass withAllSuperclassesDo: [:class |
-							class methodsDo: [:m | | sel |
-								sel := m selector asString.
-								(sel sandblockMatch: aString) ifTrue: [stream nextPut: sel]]]]]
+							class selectors sorted do: [:sel |
+								stream nextPut: sel asString]]].
+					(exactMatches, fuzzyMatches, allMatches) withoutDuplicates]
```

Comment:
I'm currently reworking this part again to add support for class-specific suggestion filters (but will take some time to submit another PR) - but eventually, the duplication is planned to be eliminated definitively. :-)